setup.py

```diff
@@ -14,23 +14,8 @@
     version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", version_file, re.M)
     version = version_match.group(1)
 
-external = Path(__file__).parent / "itables" / "external"
-if not external.is_dir():
-    external.mkdir()
-for name, url in [
-    ("jquery.min.js", "https://code.jquery.com/jquery-3.6.0.min.js"),
-    (
-        "jquery.dataTables.min.css",
-        "https://cdn.datatables.net/1.13.1/css/jquery.dataTables.min.css",
-    ),
-    (
-        "jquery.dataTables.mjs",
-        "https://cdn.datatables.net/1.12.1/js/jquery.dataTables.mjs",
-    ),
-]:
-    r = requests.get(url)
-    with open(str(external / name), "wb") as fp:
-        fp.write(r.content)
+subprocess.check_call(["npm", "install", "--prefix", "itables/dt_package"])
+subprocess.check_call(["npm", "run", "build", "--prefix", "itables/dt_package"])
```

Comment:
@mahendrapaipuri this works locally, howevver I am not sure what is the proper way to include this build step in the packaging. Would you have suggestions on how to do this? Maybe with `hatch`? Thanks!

Reply:
Yes, we can use `hatch` for this just like we are doing it for `jupytext`. I guess it is worth moving the packaging to `pyproject.toml` so that we can use same as `jupytetx`?

Reply:
That would be so much neater indeed! Would you have an example on how to call a Python function (that here will, in turn, call `npm`)? Thank you @mahendrapaipuri !

Reply:
Something like [here in jupytext](https://github.com/mwouts/jupytext/blob/1ef1dea7d1d63e418ecc093025510e7b941ade9c/pyproject.toml#L133-L163)

An "unverfied" example would be

```
[tool.hatch.build.hooks.jupyter-builder]
enable-by-default = true
dependencies = [
    "hatch-jupyter-builder>=0.5", 
]
build-function = "hatch_jupyter_builder.npm_builder"
ensured-targets = [
    "itables/dt_bundle/dt.js",
]
skip-if-exists = ["itables/dt_bundle/dt.js"]

[tool.hatch.build.hooks.jupyter-builder.build-kwargs]
path = "itables/dt_package"
build_cmd = "build"
npm = ["npm"]
```

Reply:
Thank you so much @mahendrapaipuri ! Indeed I had not realized that `npm_builder` is exactly what I was looking for. I think I am seeing this working locally.

Reply:
Hey @mahendrapaipuri , I just published a [rc](https://pypi.org/project/itables/2.0.0rc1/) to pypi, but apparently it misses `dt_bundle.js`. Can you tell what I got wrong in my `publish.yml` workflow? Thanks!

Reply:
I think you are missing `artifacts` config for `hatch`. Try something like this:

```
[tool.hatch.build.targets.sdist]
artifacts = ["itables/dt_bundle"]

[tool.hatch.build.targets.wheel]
artifacts = ["itables/dt_bundle"]
```
By default `hatch` will ignore everything in set in `.gitignore` for packaging You need explicitly [specify the artifacts](https://hatch.pypa.io/1.9/config/build/#artifacts) that are in `.gitignore` you are intending to package. Not sure if we need to set `artifacts` for both `sdist` and `wheel`. You can do tests and see the behaviour.

Reply:
I see! I will fix this tonight. Thanks @mahendrapaipuri!

Reply:
It worked! And it also solved the install on binder. Thank you @mahendrapaipuri :pray: